calabash-cucumber/lib/calabash-cucumber/automator/device_agent.rb

```diff
@@ -128,7 +129,12 @@ def swipe(options)
           end
 
           hash = query_for_coordinates(dupped_options)
-          from_point = hash[:coordinates]
+
+          from_point = if !options[:offset].nil? && options[:query].nil?
+                         options[:offset]
+                       else
+                         hash[:coordinates]
+                       end
```

Comment:
I would like to see some example `swipe` calls that are failing before this behavior is changed.

From the  [0.19.2 docs for swipe](https://github.com/calabash/calabash-ios/blob/master/calabash-cucumber/lib/calabash-cucumber/core.rb#L318):

```
By default, the gesture starts at the center of the screen.

# By default, :query is nil
@option options {String} :query (nil) if specified, the swipe will be made relative to this query.

# When query is nil, the touch point is the center of the screen
@option options {Hash} :offset (nil) optional offset to touch point. Offset supports an `:x` and `:y` key
   and causes the touch to be offset with `(x,y)` relative to the center (`center + (offset[:x], offset[:y])`).
```

Looking at the original implementation in UIA:

```
def swipe(options)
  query_action(options, :uia_swipe_offset, options)
end


https://github.com/calabash/calabash-ios/blob/develop/calabash-cucumber/lib/calabash-cucumber/automator/instruments.rb#L316
# query_action
if ui_query
  < snip >
else
  # Query was nil.  Use the offset as the initial touch point.
  # *args is options (which contains the :offset)
  # See the example below.
  self.send(action, offset, *args)
end
```

This ultimately leads to this call:

```   
    def uia_swipe_offset(offset, options)
        uia_handle_command(:swipeOffset, offset, options)
      end
```

where the behavior disappears from view - it forwarded to some JavaScript in UIA land where I will assume that it is doing the right thing given the arguments.

Here is an example:

```
# Ruby call with UIA
swipe(:left, {offset: {x: 10, y:10}, < other key/value pairs >})

# The arguments passed to the JavaScript function.
swipeOffset({x: 10, y:10}, {offset: {x: 10, y:10}, < other key/value pairs >})

# The correct call should be:
swipeOffset({< screen center >}, {offset: {x: 10, y:10}, < other key/value pairs >})
```

It seems to me that the UIA implementation is incorrect - it does not follow the documentation.

The DeviceAgent implementation is faithful to the documentation.  If query is nil, the swipe action starts on the center of the view.  If there is an offset, it is applied to the initial touch point.  When query is nil, the initial touch point is the center of the view.